libs/wire-api-federation/src/Wire/API/Federation/API/Galley.hs

```diff
@@ -220,6 +220,8 @@ ccRemoteOrigUserId cc = qualifyAs (ccCnvId cc) (ccOrigUserId cc)
 data NewRemoteConversation = NewRemoteConversation
   { -- | The conversation ID, local to the backend invoking the RPC.
     nrcConvId :: ConvId,
+    -- | The subconversation, if set.
+    nrcSubConvId :: Maybe SubConvId,
```

Comment:
Not a fan of encoding things with such ambiguity. It's a bit unintuitive that this maps to a sub-conversation if this value is present. Could we change this into a sum type instead?

Reply:
It is a sum type (`Maybe`). I'm not sure where the ambiguity is. We are using the same representation for events.

Reply:
The way we handled a need for a change like this is to use the `ConvOrSubConvId` type instead of `ConvId`, and that type can carry either of the conv ID or the full subconv ID (maybe modulo the domain). This would be a change that affects the federation API beyond MLS, so I'd recommend a separate PR to `develop` just to change these types/API and then rebase the PR to `mls` on top of it.

Reply:
I'm still not sure what we are looking to achieve. What is the problem with the current representation?

Reply:
There's nothing wrong. It will get the job done. It's just not consistent with how it was done before, but maybe we shouldn't worry about it.

Another note: the way I did such a change before is to split the request type into two: one for Proteus (which doesn't feature subconversations) and one for MLS. It could be done here too.

Reply:
It would be the [first commit](https://github.com/wireapp/wire-server/pull/2997/commits/3e6e87138fd9063653360cf16a83856e666146d8) in `develop` and [the second](https://github.com/wireapp/wire-server/pull/2997/commits/ba0b055bafc45a46df4a52e226e51659220c85f6) in `mls`. This would require `SubConversationStore` in `develop` which doesn't exist there yet.

Reply:
> It would be the [first commit](https://github.com/wireapp/wire-server/pull/2997/commits/3e6e87138fd9063653360cf16a83856e666146d8) in `develop` and [the second](https://github.com/wireapp/wire-server/pull/2997/commits/ba0b055bafc45a46df4a52e226e51659220c85f6) in `mls`. This would require `SubConversationStore` in `develop` which doesn't exist there yet.

Unfortunately, the links lead to nowhere now (I suppose due to a rebase). The split of the message sending types that I was referring to was in PR #2925.